linebacker/src/main/scala/io/chrisdavenport/linebacker/contexts/Executors.scala

```diff
@@ -37,38 +47,92 @@ object Executors {
    * 1 as to maximize the other pool for handling
    * requests or other work.
    */
-  def workStealingPool[F[_]: Sync](n: Int) =
-    executorServiceResource(workStealingPoolUnsafe(n))
+  def workStealingPool[F[_]: Sync](n: Int): Resource[F, ExecutorService] =
+    workStealingPoolC(n)(identity)
+
+  def workStealingPoolC[F[_]: Sync](n: Int)(
+      configureThreadFactory: F[ForkJoinWorkerThreadFactory] => F[ForkJoinWorkerThreadFactory]): Resource[
+    F,
+    ExecutorService] =
+    executorServiceResource(workStealingPoolUnsafe(n, configureThreadFactory))
 
   /**
    * Default Pool For Scala, optimized for forked work and then returning to a
    * main pool, generally ideal for your main event loop.
    */
-  def forkJoinPool[F[_]: Sync](n: Int) =
-    executorServiceResource(forkJoinPoolUnsafe(n))
+  def forkJoinPool[F[_]: Sync](n: Int): Resource[F, ExecutorService] =
+    forkJoinPoolC(n)(identity)
+
+  def forkJoinPoolC[F[_]: Sync](n: Int)(
+      configureThreadFactory: F[ForkJoinWorkerThreadFactory] => F[ForkJoinWorkerThreadFactory]): Resource[
+    F,
+    ExecutorService] =
+    executorServiceResource(forkJoinPoolUnsafe(n, configureThreadFactory))
 
   private def executorServiceResource[F[_]: Sync](
       f: F[ExecutorService]): Resource[F, ExecutorService] =
     Resource.make[F, ExecutorService](f)(es => Sync[F].delay(es.shutdownNow).void)
 
   object unsafe {
-    def unboundExecutorUnsafe[F[_]: Sync]: F[ExecutorService] = Sync[F].delay {
-      E.newCachedThreadPool(new ThreadFactory {
-        private val counter = new AtomicLong(0L)
-
-        def newThread(r: Runnable) = {
-          val th = new Thread(r)
-          th.setName("linebacker-thread-" + counter.getAndIncrement.toString)
-          th.setDaemon(true)
-          th
+    def unboundExecutorUnsafe[F[_]: Sync](
+        configureThreadFactory: F[ThreadFactory] => F[ThreadFactory]): F[ExecutorService] = {
```

Comment:
This is a weird function, Why is the initial `F` there and this isn't instead `ThreadFactory => F[ThreadFactory]`

Reply:
I mentioned it in the PR description :)

Reply:
Right, but my inclination, is we could rather expose a `ThreadFactory` constructor instead. Which we would then reference, as Creating values that are then ignored is in general bad practice. 

I get the feeling we're satisfying the concerns by a side-vector rather than building direct arguments to address the concerns.

Reply:
Oh, i see. Yeah, that's a good point, I'll check out the alternative.